### ISSUE TITLE
[QA] Play m3u8 playlist working from app but not from browser

### ISSUE THREAD

**Issue by manias21** [AlexPresso/VideoStation-FFMPEG-Patcher#97]
**Question**

m3u8 and ts:
-You probably came to the same test, in version dsm6.1, with VideoStation.spk.
-create a file with the extension .ts and add the example:
[example.txt](https://github.com/AlexPresso/VideoStation-FFMPEG-Patcher/files/12778773/example.txt)

-It indexes it and then with the videostation app you can play it, with the web I never managed to do it, but with the app I can.

-The question..... I can't get close to solving the dsm7, it does index it, yes, but it doesn't understand the file because the time of the file is always 0:00:00,
when it detects it in the dsm6 it is -1:00:00, so... do you have any ideas?

maybe it has something to do with it?... -protocol_whitelist file,tcp,http

By the way, I like your project

**Comment by AlexPresso** [comment]
Hello,
I'm not sure to understand the issue, you're trying to play a `m3u8` file from VideoStation ?

My understanding is: 
- you put the `m3u8` file in the VideoStation's movie directory.
- VideoStation is able to index de video
- You can play the video using the app but not from the web browser ?

**Comment by manias21** [comment]
yes.

**Comment by AlexPresso** [comment]
Okay, so this is normal behaviour.

VideoStation is not made to read `m3u8` files, it could be able to do it because it's using ffmpeg as a backend but VideoStation by itself is designed to "produce" `m3u8` playlists, not consume them.

The reason you can read these files using the DSVideo app, is because in some cases (you are in one of these cases), no transcoding occurs on the NAS itself. The movie file (in your case `m3u8` playlist) is directly sent to the TV and the "consuming" part (transcoding if necessary) happens on the TV.

The ffmpeg on the TV is able to read `m3u8` files, that's why you can read it.
You can have more infos about VideoStation's worflow here: https://github.com/AlexPresso/VideoStation-FFMPEG-Patcher/issues/88#issuecomment-1690401096 

To summarize: when using from the browser, VideoStation is just generating parameters to be used by ffmpeg to transcode a movie file (this transcoding is actually generating a `m3u8` file), that's why it won't work in your case. I think VideoStation is not designed to read `m3u8` files.

**Comment by AlexPresso** [comment]
But still, if you are using a patched version of VideoStation, you can follow the following steps to generate a issue report, si I can have a look if we can do something about it or if it's pure VideoStation limitation:

- Start the video on VideoStation, wait for the video to play and put the browser in background (keep the video playing)
- Connect through SSH ([help](https://www.synology.com/en-global/knowledgebase/DSM/tutorial/General_Setup/How_to_login_to_DSM_with_root_permission_via_SSH_Telnet))
- Switch to root user: `sudo -i`
- Run the report tool: `curl https://raw.githubusercontent.com/AlexPresso/VideoStation-FFMPEG-Patcher/main/issue-report.sh | bash`
- Copy-paste the output to a new [gist](https://gist.github.com/) and put the link here

**Comment by manias21** [comment]
I agree, all your words make sense.
The use that is processed in version 2.3.2 is formed in this way.
It is the external player used in the dsvideo app that obtains the link and processes it, but somehow in order to achieve this it has to understand the video and be able to provide service, that is why it is shown as in the image in version 2.3. 2
![dms6-2 3 2](https://github.com/AlexPresso/VideoStation-FFMPEG-Patcher/assets/92815537/02d71d18-9a46-453e-b75a-463901437914)

**Comment by manias21** [comment]
In all other versions in dsm7.x I have not succeeded.
I thought it could be because of the gstreamer, how can I force a test?

**Comment by AlexPresso** [comment]
Please send me a issue report, it will contain any error when the player tries to play the file: https://github.com/AlexPresso/VideoStation-FFMPEG-Patcher/issues/97#issuecomment-1742909041 

It may give us some clues to play with

**Comment by manias21** [comment]
can it through the... gstreamer?

dsm6
Patch Details......................................
  Is patched ? no
  Has gstreamer ? yes

dsm7
Patch Details......................................
  Is patched ? yes
  Has gstreamer ? no

[dsm6.txt](https://github.com/AlexPresso/VideoStation-FFMPEG-Patcher/files/12786125/dsm6.txt)
[dsm7.txt](https://github.com/AlexPresso/VideoStation-FFMPEG-Patcher/files/12786126/dsm7.txt)

**Comment by AlexPresso** [comment]
For DSM 6:
It seems that you're not using the patch, please patch your VideoStation version before playing the file and running the issue-report tool again.

For DSM 7:
It seems that no error files were generated by the wrapper, did you play the video before running the issue-report tool ?

**Comment by AlexPresso** [comment]
For DSM7, in this situation, either VideoStation's app has an issue or your m3u8 file is incorrect, but it seems not related to the patch

But as far as I know, there's no timecode or duration metadata in an m3u8 playlist, the slices are just played in the top bottom order

**Comment by AlexPresso** [comment]
Sorry, I can't do anything for this issue as it's not related to the patch itself, it seems to be a difference with the new version of VideoStation.

What you could do is try to convert your m3u8 playlist file to a single mp4 file, after that everything should work fine

**Comment by Kissycat** [comment]
> Sorry, I can't do anything for this issue as it's not related to the patch itself, it seems to be a difference with the new version of VideoStation.

It seems a http hls stream in a https access like this, and it can not solved by nginx return 302 or 301.
![can't play](https://dlg.lovegood.cyou/file/2/share/Screenshot_2023-11-17-14-15-08-498-edit_com.kiwibrowser.browser.jpg)

**Comment by AlexPresso** [comment]
In this case it looks like a misconfiguration with your NAS networking / ports / certificates

**Comment by Kissycat** [comment]
> In this case it looks like a misconfiguration with your NAS networking / ports / certificates

Yes, it is, caused by nginx ssl out of dsm, then I need not place cert files into dsm. 
And anyway, there is nothing else with this issue, but the hls stream, I can not find the source problem, so I finally put the cert files in..
